### ISSUE TITLE
How to run inference on custom point clouds?

### ISSUE THREAD

**Issue by anikimmel** [Pointcept/SAMPart3D#10]
Hello!

How would I run inference on custom point clouds that I want to segment?

Thanks!

**Comment by yhyang-myron** [comment]
Hi, Thanks for your interest in our work!
You can replace the coordinates and normals sampled from the mesh surface with your custom point clouds and estimated normals.

**Comment by anikimmel** [comment]
Hey! Thanks for your response! Could you by any chance direct me to which script I would be using for that to fine-tune or evaluate segmentation without labels? I was trying to figure out which script I should be using to input the .xyz files and their normals.

**Comment by yhyang-myron** [comment]
Hi, you need to change the code at [here](https://github.com/Pointcept/SAMPart3D/blob/6f889d7ae7b6a2d2e2120a75477dd455883c571a/pointcept/datasets/dataset_render_16views.py#L98). And you also need to delete the parameters like "face_index" or "inverse".

**Comment by anikimmel** [comment]
Generally, we want to train on an entire dataset. This command trains on a specific object:
sh scripts/train.sh -g 1 -d sampart3d -c sampart3d-trainmlp-render16views -n knight -o knight

How would we finetune the existing model on a new dataset? Which commands would we call/parameters to pass in?

Thanks!

**Comment by AliRezaBeigy** [comment]
I've been working with the current codebase and have encountered some issues that might be causing errors during execution. Specifically:

1. [**Definition of `offset` in `dataset_render_16views.py`**](https://github.com/Pointcept/SAMPart3D/blob/6f889d7ae7b6a2d2e2120a75477dd455883c571a/pointcept/datasets/dataset_render_16views.py#L118
)
     ```python
     offset = torch.tensor(coord.shape[0])
     ```
     This line creates a scalar tensor from an integer value.
   - In [`misc.py`](https://github.com/Pointcept/SAMPart3D/blob/6f889d7ae7b6a2d2e2120a75477dd455883c571a/pointcept/models/utils/misc.py#L6
), `torch.diff()` is applied to `offset`. However, `torch.diff()` expects an input tensor with at least one dimension, not a scalar tensor.

2. [**Type of `coord` from `sample_surface`**](https://github.com/Pointcept/SAMPart3D/blob/6f889d7ae7b6a2d2e2120a75477dd455883c571a/pointcept/datasets/dataset_render_16views.py#L102
)
     ```python
     coord, face_index, color = sample_surface(mesh, count=sample_num, sample_color=True)
     ```
     Here, `coord` is returned as a NumPy array.
   - In [`structure.py`](https://github.com/Pointcept/SAMPart3D/blob/6f889d7ae7b6a2d2e2120a75477dd455883c571a/pointcept/models/utils/structure.py#L148
), `coord` is used as follows:
     ```python
     min_coord = torch_scatter.segment_csr(self.coord, idx_ptr, reduce="min")
     ```
     The function `torch_scatter.segment_csr` expects `self.coord` to be a PyTorch tensor.

@yhyang-myron I’d appreciate it if you could confirm whether this is the latest implementation intended for use or if there have been any recent updates. Thanks in advance!
